src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+            Math_Rand_CenteredFloat(5000.0f));
+
+    if (Actor_HasNoParent(&this->actor, globalCtx) != 0) {
+        if (this->actor.params == 0x000D) {
+            this->sfxTimer1 = 0;
+            this->unk_2A6 = 1;
+            this->actionFunc = func_80AB6EB4;
+            this->actor.velocity.y = 4.0f;
+            return;
+        }
+
+        this->actor.shape.rot.z = 0;
+        this->actor.shape.rot.y = this->actor.shape.rot.z;
+        this->actor.shape.rot.x = this->actor.shape.rot.z;
+        this->actor.flags |= 1;
+        this->actionFunc = func_80AB6D08;
+    }
+    func_80AB5BF8(this, globalCtx, 2);
+}
+
+void func_80AB6D08(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->path == 0) {
+        if (!(this->actor.bgCheckFlags & 1)) {
+            return;
+        }
+
+        if (this->actor.params == 0x000E) {
+            this->unk_2AC = this->unk_2B8 = this->actor.posRot.pos.x;
+            this->unk_2B0 = this->unk_2BC = this->actor.posRot.pos.y;
+            this->unk_2B4 = this->unk_2C0 = this->actor.posRot.pos.z;
+            this->timer5 = this->timer4 = this->unk_29E = 0;
+
+            this->unk_26C[7] = this->unk_26C[5] = this->unk_26C[6] = this->unk_26C[8] = this->actor.speedXZ =
+                this->unk_2FC = this->unk_300 = 0.0f;
+
+            this->actionFunc = EnNiw_ResetAction;
+            return;
+        }
+
+        this->path = 1;
+        this->timer5 = 80;
+        this->actor.speedXZ = 0.0f;
+        this->actor.velocity.y = 4.0f;
+    } else {
+        if (this->actor.bgCheckFlags & 1) {
+            this->sfxTimer1 = 0;
+            this->actor.velocity.y = 4.0f;
+            this->unk_2A6 = 1;
+        }
+
+        if (this->timer5 == 0) {
+            this->timer6 = 100;
+            this->timer4 = 0;
+            this->path = 0;
+            this->actionFunc = func_80AB7290;
+            return;
+        }
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+        this->sfxTimer1 = 30;
+        this->path = 0;
+        this->timer4 = 30;
+        this->actor.flags &= ~1;
+        this->actor.speedXZ = 0.0f;
+        this->actionFunc = func_80AB6BF8;
+    } else {
+        if (this->timer5 >= 6) {
+            func_8002F580(&this->actor, globalCtx);
+        }
+
+        func_80AB5BF8(this, globalCtx, 2);
+    }
+}
+
+void func_80AB6EB4(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->actor.posRot.pos.y > 400.0f) {
+        Actor_Kill(&this->actor);
+    }
+
+    func_80AB5BF8(this, globalCtx, 2);
+}
+
+void func_80AB6F04(EnNiw* this, GlobalContext* globalCtx) {
+    Vec3f sp2C;
+
+    if (this->unk_2A8 != 0) {
+        EnNiw_SpawnAttackCucco(this, globalCtx);
+    }
+
+    this->actor.speedXZ = 2.0f;
+
+    if (this->actor.bgCheckFlags & 0x20) {
+        this->actor.gravity = 0.0f;
+
+        if (15.0f < this->actor.waterY) {
```

Comment:
```suggestion
        if (this->actor.waterY > 15.0f) {
```
if you're part of the consts-on-right team, otherwise not required